domains/transfer-agreements/TransferAgreements.API/API.IntegrationTests/Factories/TransferAgreementsApiWebApplicationFactory.cs

```diff
@@ -135,9 +148,27 @@ private static async Task TruncateTables(ApplicationDbContext dbContext)
     {
         var historyTable = dbContext.Model.FindEntityType(typeof(TransferAgreementHistoryEntry)).GetTableName();
         var agreementsTable = dbContext.Model.FindEntityType(typeof(TransferAgreement)).GetTableName();
+        var connectionsTable = dbContext.Model.FindEntityType(typeof(Connection)).GetTableName();
 
         await dbContext.Database.ExecuteSqlRawAsync($"TRUNCATE TABLE \"{historyTable}\"");
         await dbContext.Database.ExecuteSqlRawAsync($"TRUNCATE TABLE \"{agreementsTable}\" CASCADE");
+        await dbContext.Database.ExecuteSqlRawAsync($"TRUNCATE TABLE \"{connectionsTable}\"");
+    }
+
+    private static async Task InsertConnection(ApplicationDbContext dbContext, Connection connection)
+    {
+        var connectionTable = dbContext.Model.FindEntityType(typeof(Connection)).GetTableName();
+
+        var agreementQuery = $"INSERT INTO \"{connectionTable}\" (\"Id\", \"CompanyId\", \"CompanyTin\", \"OwnerId\") VALUES (@Id, @CompanyId, @CompanyTin, @OwnerId)";
+        var agreementFields = new[]
+        {
+            new NpgsqlParameter("Id", connection.Id),
+            new NpgsqlParameter("CompanyId", connection.CompanyId),
+            new NpgsqlParameter("CompanyTin", connection.CompanyTin),
+            new NpgsqlParameter("OwnerId", connection.OwnerId)
+        };
+
+        await dbContext.Database.ExecuteSqlRawAsync(agreementQuery, agreementFields);
```

Comment:
Why is this inserted using Raw SQL?

Reply:
Excellent question. Because that's what was already used - I've corrected it